### PR TITLE
Fix trident lightning strike on pvp disabled plots

### PR DESCRIPTION
**Pull request by Kovalskki**
#### Description: 
Fix trident lightning strike on pvp disabled plots
Known bug: Lightning Strike is disabled but sound of Channeling enchantment isnt cancelled

____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by Kovalskki** [comment]
I tried canceling the channeling voice with EntityDamagedByEntityEvent but Trident is a entity. I cant access enchantments without nbt.

**Comment by Kovalskki** [comment]
If you want, I can create a special class for this and keep the thrown spears in memory for fix the issue.

**Comment by LlmDl** [comment]
This might be something that the API/damage_events branch could fix? I haven't had a chance to look over the changes in this PR yet.

**Comment by LlmDl** [comment]
Also, I thought there was a lightning strike cause which could be useful.

File: src/com/palmergames/bukkit/towny/listeners/TownyWorldListener.java [hunk 1]
```diff
@@ -205,4 +207,18 @@ public void onPortalCreate(PortalCreateEvent event) {
 		}
 	}
 
+	@EventHandler(priority = EventPriority.NORMAL)
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/listeners/TownyWorldListener.java:210]: I haven't yet seen if this PR is necessary after the api/damage_events branch is merged but here's a replacement block for the code below, which will cover both the wilderness and non-wilderness, as well as handle worlds where Towny isn't active or when Towny doesn't load properly:

```java
	@EventHandler(priority = EventPriority.NORMAL)
	public void onLightningStrike(LightningStrikeEvent event) {
		
		if (plugin.isError()) {
			event.setCancelled(true);
			return;
		}

		if (!TownyAPI.getInstance().isTownyWorld(event.getWorld()))
			return;
		
		TownyWorld townyWorld = null;
		try {
			townyWorld = TownyUniverse.getInstance().getDataSource().getWorld(event.getLightning().getWorld().getName());
		} catch (NotRegisteredException ignored) {}
		
		if (townyWorld != null && event.getCause().equals(LightningStrikeEvent.Cause.TRIDENT)) {
			event.setCancelled(CombatUtil.preventPvP(townyWorld, TownyAPI.getInstance().getTownBlock(event.getLightning().getLocation())));
		}
	}
```

**Review by LlmDl — CHANGES_REQUESTED** [review]

**Comment by Kovalskki** [comment]
I opened issue for this https://github.com/TownyAdvanced/Towny/issues/4647#issue-776200188

**Comment by LlmDl** [comment]
Ok, will you be applying the code I gave above and re-opening this PR?

**Pull request closed without merging**